### ISSUE TITLE
Notification system in CMS - new features and other updates

### ISSUE THREAD

**Issue by troughc** [culturecreates/footlight-app#827]

**Comment by troughc** [comment]
see https://github.com/culturecreates/footlight-app/issues/789

**Comment by troughc** [comment]
@sahalali as discussed, think about a plan for:
- How to choose which releases to announce in the CMS (not all will be announced)
- How to make sure the announcement is only displayed one time for each user (once read, it is dismissed automatically)
- process for writing, reviewing/editing, and publishing the content of the messages - who is responisible for each step 
- should we use a document template ? (we think yes)

**Comment by sahalali** [comment]
@troughc I have proposed a solution for this and the same is documented [here](https://docs.google.com/document/d/1cqIU78saWpEzVJ4MPv06QW8tvggL0AXg5f-iY0_QAmU/edit)

**Comment by troughc** [comment]
@sahalali please see the figmas (in description at top) and assign this issue to the right person for an estimate. 
Thanks!
cc @saumier

**Comment by sahalali** [comment]
APIs to add:
1. POST /messages Publish message (Super admin for release messages).
2. PATCH /messages/{id}/read Read message
3. GET /messages List messages

Total backend estimate: 16 Hrs

**Comment by AbhishekPAnil** [comment]
Frontend estimates : 

1. Add the component to show notification - 4h
2. Add the default messages - 1h
3. Api integration to get the notification information ( Polling ) -5h
4. Show the notification icon on top -0.2h
5. Add routing for "Cms updates" - 1h
6. Add the CMS updates page - 5h
7. Api integration to get the cms application updates. 4h
8. Add routing for reviewing events. ? (@AbhishekPAnil Check with Caitlin if this is needed in this phase ,If yes then additional work has to be done get the details of the mentioned events)

**Comment by troughc** [comment]
@saumier suggests a design document before starting.

**Comment by dev-aravind** [comment]
Updated backend estimate:

POST /messages Publish message (Super admin for release messages). 
PATCH /messages/{id}/read Read message
GET /messages List messages - 16 hours

template design for messages - 4h
updating cms and aggregator to send notifications on certain events - 3h

Total - 23h

**Comment by sahalali** [comment]
@dev-aravind As we discussed:

1. Implement support for filtering notifications by type. This will allow us to display system updates on a separate screen, as per the design.

2. Please continue with the template approach, and let's review it eventually for notification creation. Please keep in mind that fewer DB calls should be made.

3. Enable the option to fetch notification details selectively. This will help limit API polling and avoid retrieving all details every time. In this case, if there is at least one unread notification available for the user, we can avoid polling until the user opens the notification. We only need to continue polling if there are no notifications available.
Notification details can be fetched when the user clicks on the notification button. Otherwise, we can simply return the count of unread notifications. This can be managed with a separate parameter.


cc @troughc

**Comment by dev-aravind** [comment]
@sahalali Adding one more point, the system update notifications will behave the same as the other notifications, the data will be stored individually for each user.

**Comment by troughc** [comment]
> according to a discussion between [@SyamBabu-M](https://github.com/SyamBabu-M) and [@troughc](https://github.com/troughc) , the system update changelog should be updatable and every time an update occurs, a new notification should be sent to the every user of Footlight. [@sahalali](https://github.com/sahalali) to discuss with [@troughc](https://github.com/troughc) on the changes needed.

There may be a confusion here, so I am looking forward to discussing this with you @sahalali. I also just re-read our spec for the notification system and realized we had decided on an external way to create release notes. Again looking forward to figure out the best way to do this.

**Comment by sahalali** [comment]
@dev-aravind @SyamBabu-M 

We can create a form in the user interface for the super admin to draft and publish new release notes. According to the plan, the technical lead (@sahalali) will initially draft the release notes and share them with the product owner (@troughc). This process will be carried out outside of the CMS. 

Later, the super admin will add the release notes to a form in the CMS and publish them. This action will trigger notifications to all users.

@SyamBabu-M Please have a word with @troughc or block some time to discuss the design.

**Comment by dev-aravind** [comment]
@sahalali the backend PR can be found [here](https://github.com/culturecreates/footlight-calendar-api/pull/1469).

**Comment by troughc** [comment]
Tested in chrome: 
When I click the bell, the popup flickers
To reproduce: click the bell to open the popup.


Right now there are no new notifications so i can't test much.

I tried a workflow where I sent an event for review and then sent it back to the guest user. In both cases, the notification system didn't update, but I assume that is because we haven't implemented that functionality yet.

Example screenshot for reference

<img width="1216" height="472" alt="Image" src="https://github.com/user-attachments/assets/eb265721-f1f2-4120-9ac0-26407d006708" />

**Comment by dev-aravind** [comment]
@sahalali please merge [this](https://github.com/culturecreates/footlight-calendar-api/pull/1494) PR to fix the bugs that @troughc mentioned. Thanks.

**Comment by dev-aravind** [comment]
@sahalali one more PR needed as the system updates notification seem to be calendar specific, https://github.com/culturecreates/footlight-calendar-api/pull/1498

**Comment by dev-aravind** [comment]
@troughc this happened because the revert to draft notification is sent to the user that last modified user. So when the admin modifies the event and then reverts it to draft, they become the last modifier and hence receive a notification from themselves. 

@sahalali please review and merge [this](https://github.com/culturecreates/footlight-calendar-api/pull/1509) PR so that the notification is always sent to the creator instead of the last modifier. Thanks.

**Comment by troughc** [comment]
FYI: @dev-aravind I just tested this again and the notificaitons are appearing in the correct calendars.

For discussion : I also noticed that when my guest account created an event in **tout culture**, and I reviewed and added a time as an admin - I reverted to draft and the time I added was still there. Did you fix this too? Or was the issue in Signe Laval only? (this bug was loffed in #1943 )

thanks.

**Comment by dev-aravind** [comment]
@sahalali These is one more small addition remaining in this ticket in the aggregator, we have to update it to call the messages API after every aggregator run, let me know if you want to do this in a separate ticket.

**Comment by sahalali** [comment]
@dev-aravind, 

As we discussed in our meeting yesterday, I kindly request that you document the specific scenarios in which notifications are triggered.

**Comment by AbhishekPAnil** [comment]
Tested this feature in staging by having a guest user send an event for review, then having an admin user revert it to draft.
Each time, I received the same message with the same event name and the same user.
Refer to the screenshot below:

<img width="324" height="409" alt="Image" src="https://github.com/user-attachments/assets/f0bb1efb-d8a7-4f5d-8d69-690657af2ef9" />

**Comment by dev-aravind** [comment]
@troughc this was caused by the the message template that we use to create the notification messages being modified. For example, the template for "RequestedReview" notifications is "senderName sent eventName for review" where we replace the "senderName" and "eventName" to build the notification, but unfortunately the eventName and senderName was being updated on each notification run, causing the template to be malformed.

For testing the fix, I tested using multiple guest account and admin account to check if they are in fact receiving unique, correct notifications. 


@sahalali Please review [this](https://github.com/culturecreates/footlight-calendar-api/pull/1510) PR to fix this issue.
